backend.native/compiler/ir/backend.native/src/org/jetbrains/kotlin/backend/konan/llvm/BinaryInterface.kt

```diff
@@ -150,11 +150,21 @@ private fun acyclicTypeMangler(visited: MutableSet<TypeParameterDescriptor>, typ
 private fun typeToHashString(type: IrType)
     = acyclicTypeMangler(mutableSetOf<TypeParameterDescriptor>(), type)
 
-private val FunctionDescriptor.signature: String
+internal val IrValueParameter.extensionReceiverNamePart: String
+    get() = "@${typeToHashString(this.type)}."
+
+private val IrFunction.signature: String
     get() {
-        val extensionReceiverPart = this.extensionReceiverParameter?.let { "@${typeToHashString(it.type)}." } ?: ""
+        val extensionReceiverPart = this.extensionReceiverParameter?.extensionReceiverNamePart ?: ""
         val argsPart = this.valueParameters.map {
-            "${typeToHashString(it.type)}${if (it.isVararg) "_VarArg" else ""}"
+
+        // TODO: there are clashes originating from ObjectiveC interop.
+        // kotlinx.cinterop.ObjCClassOf<T>.create(format: kotlin.String): T defined in platform.Foundation in file Foundation.kt
+        // and
+        // kotlinx.cinterop.ObjCClassOf<T>.create(string: kotlin.String): T defined in platform.Foundation in file Foundation.kt
+
+            val argName = if (this.hasObjCFactoryAnnotation || this.isObjCClassMethod()) "${it.name}:" else ""
```

Comment:
Does this cover Objective-C extension methods?

Reply:
I have added it to the pull request.